src/Server.php

```diff
@@ -186,7 +186,7 @@ public function handleConnection(ConnectionInterface $conn)
 
             $that->writeError(
                 $conn,
-                $e->getCode() !== 0 ? $e->getCode() : 400
+                $e->getCode() !== 0 ? $e->getCode() : 500
```

Comment:
I don't think any exception code should just be forwarded to the client here as a status code. Most exception codes aren't HTTP status codes.

Reply:
> Most exception codes aren't HTTP status codes.

I've not been able to figure out if/what exception codes PHP uses. But inside react/http all codes are specifically generated with the purpose of being used as HTTP return codes.

Reply:
This particular code only catches `Exception` instances from the `RequestHeaderParser` which is specifically designed to throw with HTTP status codes.